src/db.c

```diff
@@ -2224,47 +1857,48 @@ int expireIfNeeded(redisDb *db, robj *key, int flags) {
  * `DICT_OK` response is for successful expansion. However ,`DICT_ERR` response signifies failure in allocation in
  * `dictTryExpand` call and in case of `dictExpand` call it signifies no expansion was performed.
  */
-int dbExpand(const redisDb *db, uint64_t db_size, dbKeyType keyType, int try_expand) {
-    dict *d;
+static int dbExpandGeneric(kvstore *kvs, uint64_t db_size, int try_expand) {
+    int ret;
     if (server.cluster_enabled) {
         /* We don't know exact number of keys that would fall into each slot, but we can
          * approximate it, assuming even distribution, divide it by the number of slots. */
         int slots = getMyShardSlotCount();
         if (slots == 0) return C_OK;
         db_size = db_size / slots;
-
-        for (int i = 0; i < CLUSTER_SLOTS; i++) {
-            if (clusterNodeCoversSlot(getMyClusterNode(), i)) {
-                if (keyType == DB_MAIN) {
-                    d = db->dict[i];
-                } else {
-                    d = db->expires[i];
-                }
-                int result = try_expand ? dictTryExpand(d, db_size) : dictExpand(d, db_size);
-                if (try_expand && result == DICT_ERR) {
-                    serverLog(LL_WARNING, "Dict expansion failed for db type: %s, slot: %d",
-                                keyType == DB_MAIN ? "main" : "expires", i);
```

Comment:
yes, it didn't feel right calling `serverLog` from `kvstore.c`

WDYT?

Reply:
i agree, but i'm not sure how much we need them, and if we wanna still make some effort to show them, e.g. special return value from kvstoreExpand.
i.e. these are cases where the expansion failed (skipped) because of insufficient memory.

Reply:
yeah i can do that i guess
@madolson thoughts?

but @oranagra how often does it happen that we run out of memory while expanding a dict? i mean, if we're that close to OOM, why shouldn't we just panic while expanding instead of the next time someone inserts one key too much?

Reply:
it's actually not about being close to OOM, it's about someone requesting more memory than the total memory the system has (i.e. asking got 80GB on a 64GB machine), if it comes from RDB parser, we panic, and if it comes from DEBUG RELOAD, we gracefully skip it.

come to think of it then, i'm ok dropping these prints then.